x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -0,0 +1,174 @@
+    );
+  }
+
+  // delete all previous transform
+  await deleteTransforms(
+    callCluster,
+    previousInstalledTransformEsAssets.map((asset) => asset.id)
+  );
+  // install the latest dataset
+  const datasets = registryPackage.datasets;
+  if (!datasets?.length) return [];
+  const installNameSuffix = `${registryPackage.version}`;
+
+  const transformPaths = paths.filter((path) => isTransform(path));
+
+  const transformPathDatasets = datasets.reduce<TransformPathDataset[]>((acc, dataset) => {
+    transformPaths.forEach((path) => {
+      if (isDatasetTransform(path, dataset.path)) {
+        acc.push({ path, dataset });
+      }
+    });
+    return acc;
+  }, []);
+
+  const transformRefs = transformPathDatasets.reduce<EsAssetReference[]>(
+    (acc, transformPathDataset) => {
+      if (transformPathDataset) {
+        acc.push({
+          id: getTransformNameForInstallation(transformPathDataset, installNameSuffix),
+          type: ElasticsearchAssetType.transform,
+        });
+      }
+      return acc;
+    },
+    []
+  );
+
+  // get and save transform refs before installing transforms
+  await saveInstalledEsRefs(savedObjectsClient, registryPackage.name, transformRefs);
+
+  const transforms: TransformInstallation[] = transformPathDatasets.map(
+    (transformPathDataset: TransformPathDataset) => {
+      return {
+        installationName: getTransformNameForInstallation(transformPathDataset, installNameSuffix),
+        content: getAsset(transformPathDataset.path).toString('utf-8'),
+      };
+    }
+  );
+
+  const installationPromises = transforms.map(async (transform) => {
+    return installTransform({ callCluster, transform });
+  });
+
+  const installedTransforms = await Promise.all([
+    Promise.all(installationPromises),
+  ]).then((results) => results.flat());
+
+  const currentInstallation = await getInstallation({
+    savedObjectsClient,
+    pkgName: registryPackage.name,
+  });
+
+  // remove the saved object reference
+  await deleteTransformRefs(
+    savedObjectsClient,
+    currentInstallation?.installed_es || [],
+    registryPackage.name,
+    previousInstalledTransformEsAssets.map((asset) => asset.id),
+    installedTransforms.map((installed) => installed.id)
+  );
+
+  return installedTransforms;
+};
+
+const isTransform = (path: string) => {
+  const pathParts = Registry.pathParts(path);
+  return pathParts.type === ElasticsearchAssetType.transform;
+};
+
+const isDatasetTransform = (path: string, datasetName: string) => {
+  const pathParts = Registry.pathParts(path);
+  return (
+    !path.endsWith('/') &&
+    pathParts.type === ElasticsearchAssetType.transform &&
+    pathParts.dataset !== undefined &&
+    datasetName === pathParts.dataset
+  );
+};
+
+async function installTransform({
+  callCluster,
+  transform,
+}: {
+  callCluster: CallESAsCurrentUser;
+  transform: TransformInstallation;
+}): Promise<EsAssetReference> {
+  const callClusterParams: {
```

Comment:
I was confused at first by the name of this variable as I thought it might be something that is reused. But this is about creating the transform and as far as I can see, it is not reused. 

If we use a var, I would defined it closer to where it is used or specify it inline like you did on line 153. Otherwise reading the code is a bit confusing as I first read code that creates the transform and then deletes it, but turns out it happens in the other order. At least  I was confused at first.